presto-native-execution/presto_cpp/main/common/prometheus-metrics/PrometheusReporterTest.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PrometheusReporter.h"
+#include <gtest/gtest.h>
+
+namespace facebook::presto::prometheus {
+class PrometheusReporterTest : public testing::Test {
+  void SetUp() override {}
+
+  void TearDown() override {}
+};
+
+TEST_F(PrometheusReporterTest, testAllMetrics) {
+  ::prometheus::Labels testLabels(
+      {{"cluster", "test_cluster"}, {"worker", "test_worker_pod"}});
+  std::string labelsSerialized =
+      ("cluster=\"test_cluster\",worker=\"test_worker_pod\"");
+  auto reporter = std::make_shared<PrometheusReporter>(testLabels);
+  reporter->registerMetricExportType(
+      "test.key1", facebook::velox::StatType::COUNT);
+  reporter->registerMetricExportType(
+      "test.key2", facebook::velox::StatType::AVG);
+  reporter->registerMetricExportType(
+      "test.key3", facebook::velox::StatType::SUM);
+  reporter->registerMetricExportType(
+      "test.key4", facebook::velox::StatType::RATE);
+  EXPECT_EQ(
+      facebook::velox::StatType::COUNT,
+      reporter->registeredMetrics_.find("test.key1")->second);
+  EXPECT_EQ(
+      facebook::velox::StatType::AVG,
+      reporter->registeredMetrics_.find("test.key2")->second);
+  EXPECT_EQ(
+      facebook::velox::StatType::SUM,
+      reporter->registeredMetrics_.find("test.key3")->second);
+  EXPECT_EQ(
+      facebook::velox::StatType::RATE,
+      reporter->registeredMetrics_.find("test.key4")->second);
+
+  std::vector<size_t> testData = {10, 11, 15};
+  for (auto i : testData) {
+    reporter->addMetricValue("test.key1", i);
+    reporter->addMetricValue("test.key2", i + 1000);
+  }
+  // Uses default value of 1 for second parameter.
+  reporter->addMetricValue("test.key1");
+  prometheus::PrometheusSerializer serializer(prometheus::Labels{
+      {"cluster", "test_cluster"}, {"worker", "test_worker_pod"}});
+  auto fullSerializedResult = reporter->getSerializedMetrics();
+  std::vector<std::string> expected = {
+      "# TYPE test_key1 counter",
+      "test_key1{" + labelsSerialized + "} 37",
+      "# TYPE test_key2 gauge",
+      "test_key2{" + labelsSerialized + "} 1015",
+      "# TYPE test_key3 gauge",
+      "test_key3{" + labelsSerialized + "} 0",
+      "# TYPE test_key4 gauge",
+      "test_key4{" + labelsSerialized + "} 0"};
+
+  auto verifySerializedResult = [&](std::string& fullSerializedResult,
```

Comment:
I feel you should send a const& here and make a copy in the verify routine. It makes it more readable.

Reply:
Done.